### PR TITLE
KBs for Healing Infusion + Rally Hive. Healing Infusion code improvement

### PR DESCRIPTION
**Pull request by Surrealaser**
## About The Pull Request

1. Adds hotkey for Healing Infusion and Hive Rally

2. Improves code for Healing Infusion; hopefully stops inconsistent buff SFX behaviour reported by some.

## Why It's Good For The Game

Bugfixes.

Adds hot keys.

## Changelog
:cl:
add: Hotkeys added for Rally Hive and Healing Infusion.
fix: Possible fix for Healing Infusion SFX inconsistency.
code: Code improvement for Healing Infusion.
/:cl:

### CODE REVIEW

File: code/datums/keybinding/xeno.dm [hunk 1]
```diff
@@ -453,18 +453,15 @@
...
 	description = "Rallies the hive to a target location."
 	keybind_signal = COMSIG_XENOABILITY_RALLY_HIVE
-*/
 
-/*
 /datum/keybinding/xeno/healing_infusion
 	name = "healing_infusion"
 	full_name = "Hivelord: Healing Infusion"
-	description = "Heals a target xeno over the next 10 seconds, restoring sunder and health."
+	description = "Heals a target xeno restoring sunder and health once per life tick for up to 5 ticks. 60 second duration."
```
Comment (TiviPlus) [inline code/datums/keybinding/xeno.dm:465]: a player will not know what a life tick is

File: code/modules/mob/living/carbon/xenomorph/castes/hivelord/abilities_hivelord.dm [hunk 2]
```diff
@@ -284,6 +284,7 @@ GLOBAL_LIST_INIT(thickenable_resin, typecacheof(list(
 	var/heal_range = HIVELORD_HEAL_RANGE
 	var/health_ticks_remaining = 0 //Buff ends whenever we run out of either health or sunder ticks, or time, whichever comes first
 	var/sunder_ticks_remaining = 0
+	var/infusion_active = FALSE //Flags whether the infusion is active or not
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/hivelord/abilities_hivelord.dm:287]: Autodoc & this is a bool not a flag

File: code/modules/mob/living/carbon/xenomorph/castes/hivelord/abilities_hivelord.dm [hunk 3]
```diff
@@ -343,10 +344,12 @@ GLOBAL_LIST_INIT(thickenable_resin, typecacheof(list(
 
 	patient.add_filter("hivelord_healing_infusion_outline", 3, list("type" = "outline", "size" = 1, "color" = COLOR_VERY_PALE_LIME_GREEN)) //Set our cool aura; also confirmation we have the buff
 
+	infusion_active = TRUE //Flag the infusion as being active
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/hivelord/abilities_hivelord.dm:347]: bool

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Review by TiviPlus — APPROVED** [review]

**Comment by tgstation-server** [comment]
Your Fix/Feature pull request delta is currently below zero (-11). Maintainers may close future Feature/Tweak/Balance PRs. Fixing issues or helping to improve the codebase will raise this score.

**Pull request merged**